### PR TITLE
Coverage for: factory/processing 

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- increase code coverage
  
## Proposed changes
- added unittests for factory/consensus

## Testing procedure
- standard system test

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: factory/processing/processComponents.go [hunk 1]
```diff
@@ -892,7 +889,7 @@ func (pcf *processComponentsFactory) indexAndReturnGenesisAccounts() (map[string
 
 	rootHash, err := pcf.state.AccountsAdapter().RootHash()
 	if err != nil {
-		return nil, err
+		return map[string]*outport.AlteredAccount{}, err
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply factory/processing/processComponents.go:892]: the output of `indexAndReturnGenesisAccounts()` is then passed to `indexGenesisBlocks` on L375 with a [log.Warn](https://github.com/multiversx/mx-chain-go/blob/more_factory_tests_p5/factory/processing/processComponents.go#L340) only on error.. then on [L1118](https://github.com/multiversx/mx-chain-go/blob/more_factory_tests_p5/factory/processing/processComponents.go#L1118) indexGenesisBlocks tries to access this map, which would panic if nil is returned

File: factory/processing/processComponents_test.go [hunk 2]
```diff
@@ -1,102 +1,1548 @@
...
+	t.Run("NewNodeRedundancy fails should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockProcessComponentsFactoryArgs()
+		netwCompStub, ok := args.Network.(*testsMocks.NetworkComponentsStub)
+		require.True(t, ok)
+		cnt := 0
+		netwCompStub.MessengerCalled = func() p2p.Messenger {
+			cnt++
+			if cnt > 8 {
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply factory/processing/processComponents_test.go]: removed this kind of tests, thus still be at 82.4% coverage on factory/processing

File: factory/processing/processComponents.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply factory/processing/processComponents.go:893]: :+1:

**Review by iulianpascalau — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed. 
@@                            Log scanner                                  @@
##                            more_factory_tests_p5                        ##
================================================================================
- Known Warnings              6
- New Warnings                1
- Known Errors                0
- New Errors                  0
- Panics                      0
================================================================================
- block hash does not match                  2501
- wrong nonce in block                       877
- miniblocks does not match                  0
- num miniblocks does not match              0
- miniblock hash does not match              0
- block bodies does not match                0
- receipts hash missmatch                    0
================================================================================
-  No jailed nodes on the testnet
================================================================================

**Pull request merged**